packages/caravan-fees/src/btcTransactionTemplate.ts

```diff
@@ -0,0 +1,506 @@
+ * This class is used to construct and manipulate Bitcoin transactions.
+ */
+export class BtcTransactionTemplate {
+  private readonly _inputs: BtcTxInputTemplate[];
+  private readonly _outputs: BtcTxOutputTemplate[];
+  private readonly _targetFeeRate: BigNumber;
+  private readonly _dustThreshold: BigNumber;
+  private readonly _network: Network;
+  private readonly _scriptType: string;
+  private readonly _requiredSigners: number;
+  private readonly _totalSigners: number;
+
+  /**
+   * Creates a new BtcTransactionTemplate instance.
+   * @param options - Configuration options for the transaction template
+   */
+  constructor(options: TransactionTemplateOptions) {
+    this._inputs = options.inputs || [];
+    this._outputs = options.outputs || [];
+    this._targetFeeRate = new BigNumber(options.targetFeeRate);
+    this._dustThreshold = new BigNumber(
+      options.dustThreshold || DEFAULT_DUST_THRESHOLD_IN_SATS,
+    );
+    this._network = options.network;
+    this._scriptType = options.scriptType;
+    this._requiredSigners = options.requiredSigners || 1;
+    this._totalSigners = options.totalSigners || 1;
+  }
+
+  /**
+   * Creates a BtcTransactionTemplate from a raw PSBT hex string.
+   * This method parses the PSBT, extracts input and output information,
+   * and creates a new BtcTransactionTemplate instance.
+   *
+   * @param psbtHex - The raw PSBT hex string
+   * @param options - Additional options for creating the template
+   * @returns A new BtcTransactionTemplate instance
+   * @throws Error if PSBT parsing fails or required information is missing
+   */
+  static rawPsbt(
+    psbtHex: string,
+    options: Omit<TransactionTemplateOptions, "inputs" | "outputs">,
+  ): BtcTransactionTemplate {
+    const psbt = initializePsbt(psbtHex);
+    const inputs: BtcTxInputTemplate[] = [];
+    const outputs: BtcTxOutputTemplate[] = [];
+
+    // Process inputs
+    for (let i = 0; i < psbt.PSBT_GLOBAL_INPUT_COUNT; i++) {
+      const txid = psbt.PSBT_IN_PREVIOUS_TXID[i];
+      const vout = psbt.PSBT_IN_OUTPUT_INDEX[i];
+
+      if (!txid || vout === undefined) {
+        throw new Error(`Missing txid or vout for input ${i}`);
+      }
+
+      const input = new BtcTxInputTemplate({
+        txid: Buffer.from(txid, "hex").reverse().toString("hex"),
+        vout,
+      });
+
+      let amountSats: string;
+      const witnessUtxo = psbt.PSBT_IN_WITNESS_UTXO[i];
+      const nonWitnessUtxo = psbt.PSBT_IN_NON_WITNESS_UTXO[i];
+
+      if (witnessUtxo) {
+        // add amount
+        amountSats = parseWitnessUtxoValue(witnessUtxo, i).toString();
+
+        // Parse the witness UTXO
+        const witnessUtxoBuffer = Buffer.from(witnessUtxo, "hex");
+        const value = witnessUtxoBuffer.readUInt32LE(0); // Read the first 4 bytes as the value
+        const script = witnessUtxoBuffer.slice(4); // The rest is the script
+        input.setWitnessUtxo({ script, value });
+      } else if (nonWitnessUtxo) {
```

Comment:
Ahh ... so I tried to think of how can we handle this .... I was not able to think of a good approach to handle this case ...
Can you please suggest any possible solutions that could work here ?

Reply:
I guess you're just using this for parsing amountSats so it might be fine. I'm just going to mark this as resolved, but its' something to keep an eye on when implementing into the actual caravan coordinator.